pkg/query/selection.go

```diff
@@ -109,13 +113,35 @@ type Criterion struct {
 	RightOp []string
 	// Type is the type of the query
 	Type CriterionType
+	// TemplateParameters contains the template parameters of Criterion supporting sub-queries
+	TemplateParameters TemplateParameters
 }
 
 // ByField constructs a new criterion for field querying
 func ByField(operator Operator, leftOp string, rightOp ...string) Criterion {
 	return NewCriterion(leftOp, operator, rightOp, FieldQuery)
 }
 
+func ByIdNotExistWithTemplateParameters(subQuery string, templateParameters TemplateParameters) Criterion {
+	criterion := ByIdNotExist(subQuery)
+	criterion.addTemplateParams(templateParameters)
+	return criterion
+}
+
+func ByIdNotExist(subQuery string) Criterion {
+	return NewCriterion("", NotExistsSubquery, []string{subQuery}, ExistQuery)
+}
+
+func ByIdExistWithTemplateParameters(subQuery string, templateParameters TemplateParameters) Criterion {
```

Comment:
usually the function with less parameters call the one with more parameters using nil for the missing params, and not the other way around, so you only have one function with actual logic.

Reply:
Ok, got it.